analyses/molecular-subtyping-EPN/02_ependymoma_generate_all_data.py

```diff
@@ -0,0 +1,125 @@
+EPN_notebook = pd.read_csv(args.notebook, sep="\t")
+
+# This function takes in CNA dataframe along  with chromosomal arm
+# and if should be a loss/gain in the results and returns a boolean value  accordingly
+def DNA_samples_fill_df(row, CNA, arm, loss_gain):
+    if row["Kids_First_Biospecimen_ID_DNA"] is np.nan:
+        return(0)
+    CNA_value = CNA.loc[arm, row["Kids_First_Biospecimen_ID_DNA"]]
+    if CNA_value<0 and loss_gain=="loss":
+       return(1)
+    elif loss_gain=="gain" and CNA_value>0:
+       return(1)
+    else:
+       return(0)
+
+# Function to generate Z-scores column for every gene 
+def fill_df_with_fpkm_zscores(df, fpkmdf, gene_name):
+        zscore_list = stats.zscore(np.array(df.apply(lambda x: fpkmdf.loc[gene_name, x["Kids_First_Biospecimen_ID_RNA"]], axis=1)))
+        column_name = gene_name + "_expr_zscore"
+        df[column_name] = pd.Series(zscore_list)
+        return(df)
+
+
+# This  function takes a dataframe whose values need to be  used for final 
+# EPN_notebook and based on columnname, the corresponding value is returned  
+def  fill_df_with_rnaresults(row, RNA_df, columnname):
+	fusion_value = RNA_df.loc[row["Kids_First_Biospecimen_ID_RNA"],  columnname]
+	return(fusion_value)
+
+# Filling up  dataframe with broad CNA values from GISTIC
+EPN_notebook["1q_loss"] = EPN_notebook.apply(DNA_samples_fill_df, axis = 1, args = (CNA, "1q", "loss"))
+EPN_notebook["9p_loss"] = EPN_notebook.apply(DNA_samples_fill_df, axis = 1, args = (CNA, "9p", "loss"))
+EPN_notebook["9q_loss"] = EPN_notebook.apply(DNA_samples_fill_df, axis = 1, args = (CNA, "9q", "loss"))
+EPN_notebook["6p_loss"] = EPN_notebook.apply(DNA_samples_fill_df, axis = 1, args = (CNA, "6p", "loss"))
+EPN_notebook["6q_loss"] = EPN_notebook.apply(DNA_samples_fill_df, axis = 1, args = (CNA, "6q", "loss"))
+EPN_notebook["11q_loss"] = EPN_notebook.apply(DNA_samples_fill_df, axis = 1, args = (CNA, "11q", "loss"))
+EPN_notebook["11q_gain"] = EPN_notebook.apply(DNA_samples_fill_df, axis = 1, args = (CNA, "11q", "gain"))
+
+# Adding GSEA score to the  dataframe 
+#EPN_notebook["NFKB_pathway_GSEAscore"] = EPN_notebook.apply(lambda x: gsva_NFKB.loc[x["Kids_First_Biospecimen_ID_RNA"], "gsea_score"], axis =1)
+EPN_notebook["NFKB_pathway_GSEAscore"] = EPN_notebook.apply(fill_df_with_rnaresults, axis=1, args =  (gsva_NFKB, "gsea_score"))
+
+fusions_list = ["C11orf95--RELA", "LTBP3--RELA", "PTEN--TAS2R1",  "C11orf95--YAP1", "YAP1--MAMLD1", "YAP1--FAM118B", "C11orf95--MAML2"]
+for every_fusion in fusions_list:
+	EPN_notebook[every_fusion] = EPN_notebook.apply(fill_df_with_rnaresults,  axis=1, args = (fusion, every_fusion))
+
+
+# Adding breakpoints density for chromosomal instability  to the dataframe
+EPN_notebook["breaks_density-chromosomal_instability"] = EPN_notebook.apply(lambda x: breakpoint_density.loc[x["Kids_First_Biospecimen_ID_DNA"], "breaks_density"]
+        if x["Kids_First_Biospecimen_ID_DNA"] is not np.nan  else "NA", axis=1)
```

Comment:
The file we were using here `../chromosomal-instability/breakpoint-data/union_of_breaks_densities.tsv` has been removed from that analysis (which is why CI is currently failing). There are still files for CNV and SV separately at `../chromosomal-instability/breakpoint-data/cnv-breaks-densities.tsv` and `../chromosomal-instability/breakpoint-data/sv-breaks-densities.tsv`, respectively. Having two data columns for chromosomal instability seems reasonable at this stage, and they can perhaps be useful for determining reliability of this measure.